src/coreclr/nativeaot/Runtime/inc/varint.h

```diff
@@ -1,82 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-class VarInt
-{
-public:
-    static uint32_t ReadUnsigned(PTR_uint8_t & pbEncoding)
-    {
-        uintptr_t lengthBits = *pbEncoding & 0x0F;
-        size_t  negLength = s_negLengthTab[lengthBits];
-        uintptr_t shift = s_shiftTab[lengthBits];
-        uint32_t result = *(PTR_uint32_t)(pbEncoding - negLength - 4);
```

Comment:
This was tricky branch-free implementation to squeeze a few extra cycles. Exception handling is the only place where it was used.

I have replaced it with clone of much more straightforward managed implementation. We can afford to spend a few cycles on decoding EH clauses...